### PR TITLE
Bump coredns and update resources

### PR DESCRIPTION
**Pull request by brandond**
* Bump coredns chart version and image tag
* Increase autoscaler resources so that it isn't killed for exceeding limits

Related to:
* https://github.com/rancher/rke2/issues/2175

### CODE REVIEW

File: packages/rke2-coredns/generated-changes/patch/Chart.yaml.patch [hunk 1]
```diff
@@ -1,14 +1,5 @@
 --- charts-original/Chart.yaml
 +++ charts/Chart.yaml
-@@ -2,7 +2,7 @@
-   artifacthub.io/changes: |
```
Comment (briandowns) [inline packages/rke2-coredns/generated-changes/patch/Chart.yaml.patch:4]: Was removing this bit intentional?

Comment (brandond) [reply packages/rke2-coredns/generated-changes/patch/Chart.yaml.patch:4]: Reviewing patches to patches... we're not patching it out, it was removed from the upstream Chart.yaml. Our only change remains the appVersion.

**Review by briandowns — APPROVED** [review]

File: packages/rke2-coredns/generated-changes/patch/values.yaml.patch [hunk 2]
```diff
@@ -123,17 +123,19 @@
...
-+      memory: "20Mi"
++      cpu: "25m"
++      memory: "16Mi"
      limits:
-       cpu: "20m"
+-      cpu: "20m"
 -      memory: "10Mi"
-+      memory: "20Mi"
++      cpu: "100m"
++      memory: "64Mi"
```
Comment (pdostal) [inline packages/rke2-coredns/generated-changes/patch/values.yaml.patch:134]: Why is the `memory` specified two times

**Review by pdostal — COMMENTED** [review]
I had similar problem in my rke2 cluster and increasing `cpu` and `memory` fixed the issue for me. Thank you for your patch!

**Pull request merged**